### PR TITLE
Json interception

### PR DESCRIPTION
**Pull request by ikelesoglu**
⭐️

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -38,6 +38,9 @@ dependencies {
     implementation Dependencies.appCompat
     implementation Dependencies.dagger
     kapt Dependencies.daggerCompiler
+    implementation Dependencies.retrofitRxJava2
```
Comment (fabiocarballo) [inline app/build.gradle]: Why is this needed on app? 🤔

File: app/build.gradle [hunk 2]
```diff
@@ -38,6 +38,9 @@ dependencies {
...
+    implementation Dependencies.converterGson
```
Comment (fabiocarballo) [inline app/build.gradle]: Same for this one!

File: app/src/androidTest/java/com/pokedroid/HomeActivityMockWebServerTest.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    @get:Rule
+    var rule = ActivityTestRule(HomeActivity::class.java)
+
+    private lateinit var mockServer: MockWebServer
+
+    @Before
+    @Throws
+    fun setUp() {
+        // Initialize mock webserver
```
Comment (fabiocarballo) [inline app/src/androidTest/java/com/pokedroid/HomeActivityMockWebServerTest.kt]: No need for these comments!

File: app/src/androidTest/java/com/pokedroid/HomeActivityMockWebServerTest.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        mockServer = MockWebServer()
+        // Start the local server
+        mockServer.start(8080)
+
```
Comment (fabiocarballo) [inline app/src/androidTest/java/com/pokedroid/HomeActivityMockWebServerTest.kt]: Remove empty line

File: app/src/androidTest/java/com/pokedroid/HomeActivityMockWebServerTest.kt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @Test
+    fun testPokemonUrls() {
+        val jsonFile = "mock.json"
+
+        val jsonFileHelper = JsonFileHelper()
```
Comment (fabiocarballo) [inline app/src/androidTest/java/com/pokedroid/HomeActivityMockWebServerTest.kt]: Maybe this could be a class global variable

File: app/src/androidTest/java/com/pokedroid/HomeActivityMockWebServerTest.kt [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+        val mockResponse = MockResponse()
+                .setResponseCode(200)
+                .setBody(jsonFileHelper.getStringFromFile(getInstrumentation().context, jsonFile))
+
+        mockServer.enqueue(mockResponse)
+
+        Espresso.onView(ViewMatchers.withId(R.id.button_id)).perform(ViewActions.click())
+        Espresso.onView(withId(R.id.pokemon_item)).check(matches(withText("mockName")))
+
```
Comment (fabiocarballo) [inline app/src/androidTest/java/com/pokedroid/HomeActivityMockWebServerTest.kt]: Remove empty line

File: app/src/androidTest/java/com/pokedroid/JsonFileHelper.kt [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package com.pokedroid
+
+import android.content.Context
+import java.io.BufferedReader
+import java.io.InputStream
+import java.io.InputStreamReader
+
+class JsonFileHelper {
```
Comment (fabiocarballo) [inline app/src/androidTest/java/com/pokedroid/JsonFileHelper.kt]: This should live under a package like `helpers`

File: app/src/androidTest/java/com/pokedroid/TestServiceModule.kt [hunk 8]
```diff
@@ -12,8 +13,13 @@ class TestServiceModule: ServiceModule() {
...
     private val myService: PokemonService = mockk()
 
+    @Provides
+    override fun provideRetrofit(): Retrofit {
+        return RetrofitClientInstance().retrofitClientInstance("http://localhost:8080/")
+    }
+
     @Provides
     override fun providePokemonService(retrofit: Retrofit): PokemonService {
-        return myService
```
Comment (fabiocarballo) [inline app/src/androidTest/java/com/pokedroid/TestServiceModule.kt]: Maybe we can actually just delete this provides method

File: mock.json [hunk 9]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (fabiocarballo) [inline mock.json]: This is duplicated

File: app/build.gradle (hunk 1 above)
Comment (ikelesoglu) [reply app/build.gradle]: it was needed for retrofit instance that you deleted :) will delete it
